packages/fx-core/templates/core/v3Migration/migrationReport.md

```diff
@@ -1,9 +1,37 @@
 # Upgrade Summary
-## Migration overview
-
-1. Upgraded projects to use Teams Toolkit version >= 5.0.0. 
-2. For changes to the file structure in the root path of a project, migrators:
-   * Moved everything in `.fx` to `teamsfx` folder with new file format.
-   * Moved `templates/appPackage` to `./appPackage` per latest tooling's requirement.
-   * Moved `templates/appPackage/aad.template.json` to `./aad.manifest.template.json` per latest tooling's requirement.
-4. For more detail changes, please refer to [this wiki](https://aka.ms/teams-toolkit-5.0-upgrade).
\ No newline at end of file
+
+Congratulations! Your project has been upgraded to work with latest Teams Toolkit. You can visit https://aka.ms/teamsfx-v5.0-guide to learn new features in Teams Toolkit.
+
+## Changes to your project
+
+1. Moved everything in `.fx` to `teamsfx` folder with new file format.
+    * Created new `app.yml` and `app.local.yml` file under `teamsfx` folder
+    * Moved content in `state.{env}.json`, `config.{env}.json` and `{env}.userdata` to `.env.{env}` under `teamsfx` folder
+2. Moved `templates/appPackage` to `./appPackage` and update placeholders in it per latest tooling's requirement.
+3. Moved `templates/appPackage/aad.template.json` to `./aad.manifest.template.json` and update placeholders in it per latest tooling's requirement.
+4. Updated `.vscode/tasks.json` and `.vscode/launch.json`.
+5. Updated `.gitignore` to ignore new files under `teamsfx` folder.
+
+For more detail changes, please refer to [this wiki](https://aka.ms/teams-toolkit-5.0-upgrade).
+
+## Changes to existing features in VS Code Teams Toolkit
+
+If you're using VS Code Teams Toolkit, there're some changes to existing features you need to aware:
+
+### Environment management
+1. All the environment files will be gitignored by default.
+2. When create new environment, you need to fill customized fields in the new `.env.{env_name}` file. Usually you need to provide values for all environment values with `CONFIG__` prefix.
+3. When create new environment, you need to manually create `templates/azure/azure.parameters.{env_name}.json` as Azure provision parameters and fill the parameter values accordingly.
+
+### Launch your app
+1. When launch your app for a remote environment, Teams Toolkit will no longer ask you to select an environment. You need to manually change environment name in `${dev:teamsAppId}` in `.vscode/launch.json` to launch your app for certain environment.
+
+### Provision SQL databases
+1. When you provision a new environment, you need to provide values for `STATE__FX_RESOURCE_AZURE_SQL__ADMIN` and `SECRET_FX_RESOURCE_AZURE_SQL__ADMINPASSWORD` in `.env.{env_name}` which is required input to create SQL databases.
```

Comment:
1. When you provision a new environment, you need to provide values for `STATE__FX_RESOURCE_AZURE_SQL__ADMIN` and `SECRET_FX_RESOURCE_AZURE_SQL__ADMINPASSWORD` in `.env.{env_name}` which are required inputs for creating SQL databases.